src/fiber.cr

```diff
@@ -116,32 +116,29 @@ class Fiber
     Crystal::Scheduler.reschedule
   end
 
-  def resume : Nil
-    current, Thread.current.current_fiber = Thread.current.current_fiber, self
-    GC.stack_bottom = @stack_bottom
-    Fiber.swapcontext(pointerof(current.@stack_top), @stack_top)
+  def self.current
+    Crystal::Scheduler.current_fiber
   end
 
-  def sleep(time : Time::Span)
-    event = @resume_event ||= Crystal::EventLoop.create_resume_event(self)
-    event.add(time)
-    Crystal::Scheduler.reschedule
+  def resume : Nil
+    Crystal::Scheduler.resume(self)
   end
 
-  def sleep(time : Number)
-    sleep(time.seconds)
+  # :nodoc:
+  def resume_event
+    @resume_event ||= Crystal::EventLoop.create_resume_event(self)
   end
 
-  def yield
-    sleep(0)
+  def self.sleep(time : Number)
```

Comment:
Perhaps remove these methods and use `Crystal::Scheduler` directly from the top-level sleep?